macros/regexp_count.sql

```diff
@@ -0,0 +1,18 @@
+{%- macro regexp_count(value, pattern) -%}
+    {# counts how many instances of `pattern` in `value`
+       ARGS:
+         - value (string) the subject to be searched
+         - pattern (string) the regex pattern to search for
+       RETURNS: An integer count of patterns in value
+    #}
+    {%- if target.type ==  'postgres' -%} 
+        (SELECT count(values)::int from regexp_matches({{value}},{{pattern}} , 'g') values)
+    {%- elif target.type == 'bigquery' -%}
+        (SELECT array_length(regexp_extract_all({{value}}, r{{pattern}})))
+    {%- elif target.type == 'snowflake' -%}
+        REGEXP_COUNT({{value}},{{pattern}})
```

Comment:
this wont work for all regex cases. if you have a regex with the `\` character it will fail for snowflake. Perhaps implementing an xdb helper macro that will escape the `pattern` string would work?

in this macro you would conform the param like so `{%- set pattern = escape_regex_pattern(pattern) -%}`. then `escape_regex_pattern()` could be tested as well?

Reply:
:boom: updated, good call. actually merged everything into the `regexp.sql` module so the different macros live in the same file (makes it much easier to read IMO). updated it to use the escape macro, and added a little docs there too 👍